 1) Ejabberd 1.1.2 changed error message, which is sent if the server
    doesn't understand received query, from <feature-not-implemented/>
    to <service-unavailable>. While it's recommended by XMPP RFC, it
    breaks some Jabber clients (centericq is known to fail after the
    change). The patch restores old error message.

Index: ejabberd-1.1.2/src/ejabberd_sm.erl
===================================================================
--- ejabberd-1.1.2.orig/src/ejabberd_sm.erl	2007-08-17 15:30:40.000000000 +0400
+++ ejabberd-1.1.2/src/ejabberd_sm.erl	2007-08-17 15:33:34.000000000 +0400
@@ -570,7 +570,7 @@
 					  From, To, IQ);
 		[] ->
 		    Err = jlib:make_error_reply(
-			    Packet, ?ERR_SERVICE_UNAVAILABLE),
+			    Packet, ?ERR_FEATURE_NOT_IMPLEMENTED),
 		    ejabberd_router:route(To, From, Err)
 	    end;
 	reply ->
